components/Checkboxes.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { ChangeEvent, useEffect, useState } from 'react'
+import { v4 } from 'uuid'
+
+import styles from '../styles/Checkboxes.module.css'
+
+export type CheckboxesProps = {
+  disabled?: boolean
+  options: string[]
+  updateCallback?: (update: string) => void
+}
+const Checkboxes = ({
+  disabled = false,
+  options,
+  updateCallback
+}: CheckboxesProps) => {
+  const [selectedOptions, updateSelectedOptions] = useState(
+    options.map(() => false)
+  )
+
+  useEffect(
+    () => {
+      if (!updateCallback) return
+      updateCallback(selectedOptions.toString())
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [selectedOptions]
+  )
+
+  const uuid = v4()
```

Comment:
hmmm. Does this cause the id to change on every rerender? Like when a box is checked and unchecked?